src/workerd/api/queue.c++

```diff
@@ -532,9 +532,14 @@ kj::Promise<WorkerInterface::CustomEvent::Result> QueueCustomEventImpl::run(
   }
 
   KJ_IF_SOME(t, incomingRequest->getWorkerTracer()) {
-    t.setEventInfo(context.now(), tracing::QueueEventInfo(kj::mv(queueName), batchSize));
+    t.setEventInfo(context.now(), tracing::QueueEventInfo(kj::str(queueName), batchSize));
   }
 
+  context.getMetrics().reportTailEvent(context, [&] {
+    return tracing::Onset(tracing::QueueEventInfo(kj::mv(queueName), batchSize),
+        tracing::Onset::WorkerInfo{}, kj::none);
+  });
```

Comment:
For reviewers: The `getWorkerTracer()` block above is for the old style tail workers. The `reportTailEvent` is for streaming tail workers